algebird-core/src/main/scala/com/twitter/algebird/MapAlgebra.scala

```diff
@@ -115,33 +115,23 @@ class ScMapMonoid[K, V](implicit semigroup: Semigroup[V]) extends GenericMapMono
     new MutableBackedMap(mut)
 }
 
-private[this] class MutableBackedMap[K, V](val backingMap: MMap[K, V])
-    extends Map[K, V]
-    with java.io.Serializable {
-  override def get(key: K): Option[V] = backingMap.get(key)
-
-  override def iterator: Iterator[(K, V)] = backingMap.iterator
-
-  override def +[B1 >: V](kv: (K, B1)): Map[K, B1] = backingMap.toMap + kv
-
-  override def -(key: K): Map[K, V] = backingMap.toMap - key
-}
-
 /**
  * You can think of this as a Sparse vector group
  */
 class MapGroup[K, V](implicit val group: Group[V]) extends MapMonoid[K, V]()(group) with Group[Map[K, V]] {
-  override def negate(kv: Map[K, V]): Map[K, V] = kv.mapValues { v =>
-    group.negate(v)
-  }
+  override def negate(kv: Map[K, V]): Map[K, V] =
+    kv.mapValues { v =>
```

Comment:
can we just do `kv.iterator.map { ... }.toMap` so people don't have to have the `mapValues` anxiety when they read the code?

Reply:
👍